test-data/unit/check-expressions.test

```diff
@@ -1525,9 +1525,7 @@ from typing import Iterator
 def f() -> Iterator[int]:
     yield 5
 def g() -> Iterator[int]:
-    a = yield from f()
-[out]
-main:5: error: Function does not return a value
+    a = yield from f()  # E: Expression evaluates to "None"
```

Comment:
I don't think this message is better, it is still not clear what is the problematic expression. With previous wording, it may be better to add a function name.

Reply:
But it's not necessarily a named function. I changed it to `Expression "yield from <iterator>" does not return a value`. Is that better?